### ISSUE TITLE
[rt.openssl.org #4631] Another DTLS1_BAD_VER regression in 1.0.2

### ISSUE THREAD

**Issue by richsalz** [openssl/openssl#2506]
Migrated from [rt.openssl.org#4631](https://rt.openssl.org/Ticket/Display.html?id=4631) (status was 'new')

Requestors:
* dwmw2@infradead.org

Attachments:
* [0001-Fix-SSL_export_keying_material-for-DTLS1_BAD_VER.patch](https://rt.openssl.org/Ticket/Attachment/70633/44419/0001-Fix-SSL_export_keying_material-for-DTLS1_BAD_VER.patch)
* [smime.p7s](https://rt.openssl.org/Ticket/Attachment/70633/44420/smime.p7s)

From dwmw2@infradead.org on 2016-07-26 11:04:55:
```
This fix and and some other regression fixes, along with a test case,
are lined up for HEAD/1.1 in Github PR #1296.

This one is also needed in 1.0.2.


-- 
David Woodhouse                            Open Source Technology Centre
David.Woodhouse@intel.com                              Intel Corporation
```

**Comment by mattcaswell** [comment]
I thought all DTLS1_BAD_VER issues were solved. Perhaps @dwmw2 can advise.

**Comment by dwmw2** [comment]
Indeed. We fixed this in 1.0.2 with commit df426c069, backported from commit c8a18468ca in master.

The test cases for DTLS1_BAD_VER, which are even in 1.0.2 now, should hopefully keep it working going forward.

**Comment by mattcaswell** [comment]
Great. Thanks. Closing.
